### ISSUE TITLE
There's an issue with threaded emails

### ISSUE THREAD

**Issue by molstrangler** [nylas/N1#1634]
I've noticed this for a while, and it's a problem with email with threads.

When view the email or the first time and its the top most email in the list ie. the newest email in my email list it display the following:

<img width="810" alt="image 1" src="https://cloud.githubusercontent.com/assets/4201518/13694393/793d783e-e76e-11e5-8834-b3d4ef4766da.png">

If I click on a different them then back again it changes how the email is display and shows a newer email, the response.  But I have easy why of working out there was a response after this email was delivered and as it;s the newest email in the list I 'wrongly' assume its the latest email.

<img width="829" alt="image 2" src="https://cloud.githubusercontent.com/assets/4201518/13694503/57756526-e76f-11e5-94e7-9bcba3c1c61e.png">

But how can this be as it's the latest email in my email list.  This is confusing and potential dangerous in that I am looking at the first display of the email and there is a response but the user does not know because:

Why is the response hidden from the user on the first email view
Why are threads listed with oldest first, maybe this needs to be changed to 'newest first'.

I can see a response further down my email list that is not included in the first email, and this is also a problem.

<img width="869" alt="image 3" src="https://cloud.githubusercontent.com/assets/4201518/13694711/80b17edc-e771-11e5-98ac-05a9c4f06d8a.png">

**Comment by bengotow** [comment]
Hey! I think you're running into intentional behavior around the expanding / collapsing of email threads. Here's how it currently works:

When you view a conversation, we expand all of the unread threads. If there are no unread threads, we expand the last message and collapse the others. This behavior results in the following:

1. The first time you view the thread, all the messages are unread, so you see the first one. The second one is most likely also there, but you'd need to scroll down to see it, and the scrollbar is hidden (activates on hover). 

2. The second time you view the thread, you've read all the messages, so N1 only expands the last message.

Does that seem right? Just let me know—we can definitely improve on this behavior and the affordances around it. It was designed after Gmail, but there must be a better way.

**Comment by molstrangler** [comment]
I think it could be implemented in a different way.  To a user (like me) who just wants to use a mail client it looks confusing and maybe I'm missing something.  Missing emails or have unread emails somewhere that I've not seen.

Also the way threads are handled should not need explanation but be obvious to the user.

I would recommend displaying the email the user clicked-on in the correct position in the email body window.  So showing collapsed emails above and below the open email.  Today you display all the emails above, but what if there are newer emails the user has not come across yet.

I think you do not include any newer emails that maybe there but the user has not opened yet.

Also for threads that span multiple days you just display the date in the collapsed email. I think you need to provide the user with more details when working with threaded emails.

Maybe if you have an indicator showing:

  o  Ordered email numbering (numbered bullet) to show the position, 
  o  Time different between date & time between sending emails
  o  And unread indicator if an email in the thread has not been read

In the email list window you do not thread the list, I think.  But you do thread the email body window.  This can also lead to some confusion.

**Comment by bengotow** [comment]
Hey! We do threading in both places, so I'm not sure that's causing the issue—could you try scrolling down the next time you're viewing an unread conversation and see if the newer emails are beneath the first expanded one? I think that the confusion part might be that we display conversations oldest => newest. It's split roughly 50/50 among mail clients, and some show newest mail at the top.

**Comment by molstrangler** [comment]
Not sure I can see the threading in the email list.  I see the email subject, who it's from, the date & time and the first line of the message.  Where's the threading?

And I see your last response in N1 but it only shows your last two responses and not my replies in-between.  You're only showing the user partial information.

**Comment by bengotow** [comment]
Sorry—I should have elaborated a bit. N1's thread list (on the left side) is like Gmail. You see one row for each conversation—the participants, a snippet of the most recent message on that thread, and the most recent date/time. When you click a conversation, you should see all the messages.

It's interesting that you don't see the replies you're making. Are you replying from within N1 or from the Github web interface? There are definitely additional email messages missing from the thread displayed in N1? Can you confirm you see them in another mail app or webmail?

**Comment by molstrangler** [comment]
I looked at the gmail interface as I haven't used it for a very long time, found your email above and clicked on it.  I didn't see any threading, then I found 'Conversation View.' that was switch off, so now I understand a little more.

So the direction for N1 is to replicate the gmail UX (in particular threading) rather than creating something new?  Apart from being OpenSource and all.

**As for the missing emails**

The replies I have missing are mainly from the various GitHub lists I'm on.  So they are not created by myself but sent from GitHub.  The emails are not missing, as they are in the mail list, just they are not included when I'm viewing the email thread.
